recipe/meta.yaml

```
@@ -40,8 +40,8 @@ requirements:
     - libitk-devel
     - libitk
     - nlohmann_json
-    - parallelproj *=cpu*  # [cuda_compiler_version == "None"]
-    - parallelproj *=cuda*  # [cuda_compiler_version != "None"]
+    - libparallelproj *=cpu*  # [cuda_compiler_version == "None"]
+    - libparallelproj *=cuda*  # [cuda_compiler_version != "None"]
```

```suggestion
    - libparallelproj =*=cpu*  # [cuda_compiler_version == "None"]
    - libparallelproj =*=cuda*  # [cuda_compiler_version != "None"]
```
This is using a weird mix between either the space version (which I don't like; ugh, semantically significant inline whitespace), and the version with `=`.

However, I think the problem is that you construct the build strings badly in parallelproj (see [files](https://anaconda.org/conda-forge/libparallelproj/files)). They shouldn't contain a period. Also you're not encoding the python version (now that it's not noarch anymore). Check how for example pytorch [does it](https://github.com/conda-forge/pytorch-cpu-feedstock/blob/main/recipe/meta.yaml#L33-L34), and bump the build number.

---

Thanks for the immediate response (very impressive)!
The build strings are historic (introduced by someone from the conda-forge team that reviewed my initial recipe).

Two things to make sure:
- since `libparallelproj` does not contain any python, I guess I can just ignore the python related parts in the build string for this output
- if the new cpu build strings starts with `cpu_*`, the cpu `run_export` should also be changed to `libparallelproj =*=cpu*`

---

> since `libparallelproj` does not contain any python, I guess I can just ignore the python related parts in the build string for this output

Yes, good point; in that respect, pytorch was not a great example, sorry. For a lib you can look at [faiss](https://github.com/conda-forge/faiss-split-feedstock/blob/main/recipe/meta.yaml#L132-L133) for example, though note that this has the cpu/cuda bit at the end (which was something that we never had homogenized; I'd stick it in the front now).




> if the new cpu build strings starts with `cpu_*`, the cpu `run_export` should also be changed to `libparallelproj =*=cpu*`

Exactly. Just noticed that the current run-export (`libparallelproj =*=h*  # [cuda_compiler_version == "None"]`) is wrong and needs to be changed anyway.

---

1. Just merged this [PR](https://github.com/conda-forge/parallelproj-feedstock/pull/33/files) where I bumped the build number and corrected the build strings and `run_export`.

2. For the STIR recipe, do we actually need the differentiation between the cuda/cpu version of `libparallelproj` as dependency? Not 100% sure right now (could be historic). But since the cuda version of `libprallelproj`depends on the virtual `__cuda` package, I thought that the build on non-cuda machines should pick up the cpu version. Is that correct?